src/nomad_simulations/model_method.py

```diff
@@ -392,6 +402,209 @@ def normalize(self, archive, logger) -> None:
+    )
+
+    perturbative_method = SubSection(sub_section=PerturbationMethod.m_def)
+
+    solver = Quantity(
+        type=MEnum(
+            'variational',
+            *solver_map.values(),
+        ),
+        default='variational',
+        description="""
+        Solvers of / approaches to the coupled cluster equations.
+        """,
+        a_eln=ELNAnnotation(component='EnumEditQuantity'),
+    )
+
+    explicit_correlation = Quantity(
+        type=MEnum('F12', 'R12', ''),
+        default='',
+        description="""
+        Explicit correlation treatment.
+        """,
+        a_eln=ELNAnnotation(component='EnumEditQuantity'),
+    )
+
+    is_unrestricted = Quantity(
+        type=bool,
+        description="""
+        Denotes whether the excitations are unrestricted or not.
+        It should be confused with `is_spin_polarized`,
+        which denotes restrictions on the ground state.
+        """,
+        a_eln=ELNAnnotation(component='BoolEditQuantity'),
+    )
+
+    is_size_extensive = Quantity(
+        type=bool,
+        description="""
+        If the Coupled Cluster method is size extensive.
+        """,
+        a_eln=ELNAnnotation(component='BoolEditQuantity'),
+    )  # this should go into a knowledge base
+
+    def check_orders(self, logger) -> bool:
+        """Perform a sanity check on the excitation and perturbation order.
+        Raise a logging error if any inconsistency is found.
+        """
+        if self.excitation_order is None:
+            logger.warning('`CoupledCluster.excitation_order` is undefined.')
+            return False
+        if len(self.excitation_order) > 1:
+            if 2 not in self.excitation_order:
+                logger.error('Coupled Cluster typically starts from doubles.')
+                return False
+        for order in (3, 4):
+            if order in self.excitation_order and order in self.perturbative_order:
+                logger.error(
+                    f'Order {order} is defined as both excitation and perturbative.'
+                )
+                return False
+        return True
+
+    def cc_to_type(self) -> None:
+        """Produce an educated guess based on the other parameters."""
+        name = 'CC'
+        # cover the basic cases
+        if 2 in self.excitation_order:
+            if 1 in self.excitation_order:
+                name += 'SD'
+            name += 'D'
+        # cover extended excitations
+        for order, abbrev in {3: 'T', 4: 'Q'}.items():
+            if order in self.excitation_order:
+                name += abbrev
+            elif order in self.perturbative_order:
+                name += f'({abbrev})'
+        # cover explicit correlation
+        if self.explicit_correlation is not None:
+            name += self.explicit_correlation
+        # cover specific solver approaches
+        if self.solver in self.map_solver:
+            name = self.solver_map[self.solver] + name
+
+    def type_to_cc(self) -> None:
```

Comment:
This kind of reaffirms my idea that `type` should be a str and here you do the checks.